### PR TITLE
User virtualselect for mutliselect in templates and in opfab code (#2868)

### PR DESCRIPTION
**Pull request by freddidierRTE**
- In release note :
  -  In chapter :  Tasks
  -  Text : #2868 Use Virtual Select component (https://sa-si-dev.github.io/virtual-select/#/) for multiselect

### CODE REVIEW

**Review by quinarygio — COMMENTED** [review]
Tests ok, style ok

File: ui/main/package.json [hunk 1]
```diff
@@ -56,6 +55,7 @@
     "ngx-chips": "3.0.0",
     "rxjs": "7.5.5",
     "svg-pan-zoom": "3.6.1",
+    "virtual-select-plugin": "^1.0.29",
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/package.json]: done

File: ui/main/src/app/modules/admin/components/editmodal/entities/edit-entity-modal.component.ts [hunk 2]
```diff
@@ -95,14 +90,14 @@ export class EditEntityModalComponent implements OnInit {
...
-        this.entitiesDropdownList.push({ id: id, itemName: itemName });
+        this.entitiesDropdownList.push({ value: id, label: itemName });
       }
     });
   }
 
   update() {
     this.cleanForm();
-    this.parents.setValue(this.parents.value.map(entity => entity.id));
+    //this.parents.setValue(this.parents.value.map(entity => entity.id));
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/src/app/modules/admin/components/editmodal/entities/edit-entity-modal.component.ts]: done

File: ui/main/src/app/modules/externaldevicesconfiguration/editModal/externaldevicesconfiguration-modal.component.ts [hunk 3]
```diff
@@ -94,9 +91,10 @@ export class ExternaldevicesconfigurationModalComponent implements OnInit {
   }
 
   formToUserConfig() {
+    console.log("this.externalDeviceIds.value=",this.externalDeviceIds.value);
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/src/app/modules/externaldevicesconfiguration/editModal/externaldevicesconfiguration-modal.component.ts]: done

File: ui/main/src/app/modules/share/archives-logging-filters/archives-logging-filters.component.ts [hunk 4]
```diff
@@ -203,24 +202,22 @@ export class ArchivesLoggingFiltersComponent implements OnInit, OnDestroy {
...
-            if (processGroup.id === '--')
-                this.processesWithoutProcessGroupDropdownList.forEach(process => ids.push(process.id));
+            if (processGroup === '--')
+                this.processesWithoutProcessGroupDropdownList.forEach(process => ids.push(process.value));
             else
-                this.processesDropdownListPerProcessGroups.get(processGroup.id).forEach(process => ids.push(process.id));
+                this.processesDropdownListPerProcessGroups.get(processGroup).forEach(process => ids.push(process.value));
         });
-
+        console.log("Process group filter:",ids );
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/src/app/modules/share/archives-logging-filters/archives-logging-filters.component.ts]: done

**Comment by quinarygio** [comment]
There is a difference respect to the previous component when using the select all checkbox after filtering values:  
Filter values using Search option -> click "select all" checkbox -> all the filtered values are selected but the "select all" checkbox" is unchecked and is not possible to unselect all the filtered values.

**Review by quinarygio — CHANGES_REQUESTED** [review]

**Comment by freddidierRTE** [comment]
> There is a difference respect to the previous component when using the select all checkbox after filtering values: Filter values using Search option -> click "select all" checkbox -> all the filtered values are selected but the "select all" checkbox" is unchecked and is not possible to unselect all the filtered values.

An issue has been post on the topic : https://github.com/sa-si-dev/virtual-select/issues/151

File: ui/main/src/app/modules/archives/archives.module.ts [hunk 5]
```diff
@@ -30,7 +29,6 @@ import {ArchivesLoggingFiltersModule} from "../share/archives-logging-filters/ar
     TranslateModule,
     NgbModule,
     DatetimeFilterModule,
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/src/app/modules/archives/archives.module.ts:31]: done

File: ui/main/src/app/modules/monitoring/monitoring.module.ts [hunk 6]
```diff
@@ -43,7 +43,8 @@ import {ResponsesCellRendererComponent} from './components/cell-renderers/respon
         , NgbModule
         , DatetimeFilterModule
         , AppRoutingModule
-        , MultiFilterModule, TimelineButtonsModule, AgGridModule
+        , MultiSelectModule
+        , TimelineButtonsModule, AgGridModule
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/src/app/modules/monitoring/monitoring.module.ts:47]: Done

File: ui/main/src/app/services/processes.service.ts [hunk 7]
```diff
@@ -276,14 +282,31 @@ export class ProcessesService {
         return processesPerProcessGroups;
     }
 
-    public getProcessesWithoutProcessGroup(processesFilter?: string[]): any[] {
+    public getProcessesPerProcessGroupsNew(processesFilter?: string[]): Map<any, any> {
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/src/app/services/processes.service.ts]: done

**Review by quinarygio — APPROVED** [review]

**Pull request merged**